ddtrace/debugging/_origin/span.py

```diff
@@ -0,0 +1,267 @@
+from ddtrace.span import Span
+
+
+def frame_stack(frame: FrameType) -> t.Iterator[FrameType]:
+    _frame: t.Optional[FrameType] = frame
+    while _frame is not None:
+        yield _frame
+        _frame = _frame.f_back
+
+
+def wrap_entrypoint(f: t.Callable) -> None:
+    if not _isinstance(f, FunctionType):
+        return
+
+    _f = t.cast(FunctionType, f)
+    if not EntrySpanWrappingContext.is_wrapped(_f):
+        EntrySpanWrappingContext(_f).wrap()
+
+
+@dataclass
+class EntrySpanProbe(LogFunctionProbe):
+    __span_class__ = "entry"
+
+    @classmethod
+    def build(cls, name: str, module: str, function: str) -> "EntrySpanProbe":
+        message = f"{cls.__span_class__} span info for {name}, in {module}, in function {function}"
+
+        return cls(
+            probe_id=str(uuid.uuid4()),
+            version=0,
+            tags={},
+            module=module,
+            func_qname=function,
+            evaluate_at=ProbeEvaluateTimingForMethod.ENTER,
+            template=message,
+            segments=[LiteralTemplateSegment(message)],
+            take_snapshot=True,
+            limits=DEFAULT_CAPTURE_LIMITS,
+            condition=None,
+            condition_error_rate=0.0,
+            rate=float("inf"),
+        )
+
+
+@dataclass
+class ExitSpanProbe(LogLineProbe):
+    __span_class__ = "exit"
+
+    @classmethod
+    def build(cls, name: str, filename: str, line: int) -> "ExitSpanProbe":
+        message = f"{cls.__span_class__} span info for {name}, in {filename}, at {line}"
+
+        return cls(
+            probe_id=str(uuid.uuid4()),
+            version=0,
+            tags={},
+            source_file=filename,
+            line=line,
+            template=message,
+            segments=[LiteralTemplateSegment(message)],
+            take_snapshot=True,
+            limits=DEFAULT_CAPTURE_LIMITS,
+            condition=None,
+            condition_error_rate=0.0,
+            rate=float("inf"),
+        )
+
+    @classmethod
+    def from_frame(cls, frame: FrameType) -> "ExitSpanProbe":
+        code = frame.f_code
+        return t.cast(
+            ExitSpanProbe,
+            cls.build(
+                name=code.co_qualname if sys.version_info >= (3, 11) else code.co_name,  # type: ignore[attr-defined]
+                filename=str(Path(code.co_filename).resolve()),
+                line=frame.f_lineno,
+            ),
+        )
+
+
+@dataclass
+class EntrySpanLocation:
+    name: str
+    start_line: int
+    end_line: int
+    file: str
+    module: str
+    probe: EntrySpanProbe
+
+
+class EntrySpanWrappingContext(WrappingContext):
+    def __init__(self, f):
+        super().__init__(f)
+
+        lines = linenos(f)
+        start_line = min(lines)
+        filename = str(Path(f.__code__.co_filename).resolve())
+        name = f.__qualname__
+        module = f.__module__
+        self.location = EntrySpanLocation(
+            name=name,
+            start_line=start_line,
```

Comment:
we no longer require start_line and end_lines. we can just put line number for entry and we use code tokenization now to find start and end lines.

So can remove this and no longer extract lines from the code.